pydough/sqlglot/transform_bindings/base_transform_bindings.py

```diff
@@ -0,0 +1,1389 @@
+"""
+Definition of binding infrastructure that maps PyDough operators to
+implementations of how to convert them to SQLGlot expressions
+"""
+
+__all__ = ["BaseTransformBindings"]
+
+import re
+
+import sqlglot.expressions as sqlglot_expressions
+from sqlglot.expressions import Concat
+from sqlglot.expressions import Expression as SQLGlotExpression
+
+import pydough.pydough_operators as pydop
+from pydough.configs import DayOfWeek, PyDoughConfigs
+from pydough.types import BooleanType, Int64Type, PyDoughType, StringType
+
+from .sqlglot_transform_utils import (
+    DateTimeUnit,
+    apply_parens,
+    current_ts_pattern,
+    offset_pattern,
+    pad_helper,
+    positive_index,
+    trunc_pattern,
+)
+
+
+class BaseTransformBindings:
+    """
+    The base class for converting function calls from relational expressions
+    into the SQLGlot AST. This class is used for generic ANSI SQL, while
+    subclasses can override its methods to provide dialect-specific changes.
+    """
+
+    def __init__(self, configs: PyDoughConfigs):
+        self._configs = configs
+
+    @property
+    def configs(self) -> PyDoughConfigs:
+        """
+        The PyDough configuration settings used during the SQLGlot conversion.
+        """
+        return self._configs
+
+    @property
+    def dialect_start_of_week(self) -> DayOfWeek:
+        """
+        Which day of the week is considered the start of the week within the
+        SQL dialect. Individual dialects may override this.
+        """
+        return DayOfWeek.SUNDAY
+
+    @property
+    def start_of_week_offset(self) -> int:
+        """
+        The number of days to add to the start of the week within the
+        SQL dialect to obtain the start of week referenced by the configs.
+        """
+        dows: list[DayOfWeek] = [
+            DayOfWeek.SUNDAY,
+            DayOfWeek.MONDAY,
+            DayOfWeek.TUESDAY,
+            DayOfWeek.WEDNESDAY,
+            DayOfWeek.THURSDAY,
+            DayOfWeek.FRIDAY,
+            DayOfWeek.SATURDAY,
+        ]
```

Comment:
NIT: this could be shortened - `dows: list[DayOfWeek] = list(DayOfWeek)`, unless you wanted it to be explicit.